### ISSUE TITLE
Disable no-shadow in favour of @typescript-eslint/no-shadow

### ISSUE THREAD

**Issue by brachkow** [fandsdev/eslint-config-fans#25]
Hi! We have this issue in our config — https://github.com/typescript-eslint/typescript-eslint/issues/2483

For example if you import component named «Password» and enum that contains same word like Enum.Password, this will show error which is senseless as enums are namespaced.

Fix is to do this:

```
"no-shadow": "off",
"@typescript-eslint/no-shadow": "error"
```

**Comment by brachkow** [comment]
I tested this setup, and it doesn't work. Error is still reported for Enums.

**Comment by brachkow** [comment]
Here is an answer why this rule works that way https://eslint.org/docs/latest/rules/no-shadow#why-does-the-rule-report-on-enum-members-that-share-the-same-name-as-a-variable-in-a-parent-scope
